src/components/controls/buttons/DropdownButton.js

```diff
@@ -0,0 +1,139 @@
+import React, { Children } from 'react';
+import PropTypes from 'prop-types';
+import Button from '../../controls/buttons/Button';
+import RootCloseWrapper from 'react-overlays/lib/RootCloseWrapper';
+import styled, { ThemeProvider } from 'styled-components';
+import defaultTheme from '../../theme/defaultTheme';
+
+const Caret = styled.span`
+  display: inline-block;
+  width: 0;
+  height: 0;
+  margin-left: 5px;
+  vertical-align: middle;
+  border-top: 4px dashed;
+  border-right: 4px solid transparent;
+  border-left: 4px solid transparent;
+`;
+
+const ButtonPanel = styled.div`
+  display: ${props => (props.isOpen ? 'block' : 'none')};
+  margin-top: 3px;
+  position: absolute;
+  border: 1px solid ${props => props.theme.colors.grayLight};
+  background-color: ${props => props.theme.colors.white};
+`;
+
+const ButtonPanelChildrenContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const StyledButtonLink = styled(Button)`
+  padding: 10px 20px;
+  color: black;
+  text-decoration: none;
+  text-align: left;
+  margin-bottom: 0px;
+  &:hover {
+    background-color: ${props => props.theme.colors.grayLighter};
+  }
+`;
+
+class DropdownButton extends React.Component {
+  constructor(props) {
+    super();
+    this.state = {
+      buttonValue: props.buttonValue,
+      isOpen: false
+    };
+  }
+
+  toggleDropdown = () => {
+    this.setState(previousState => ({ isOpen: !previousState.isOpen }));
+  };
+
+  closeDropdown = () => {
+    if (this.state.isOpen) {
+      this.setState({ isOpen: false });
+    }
+  };
+
+  handleDropdownItemClick = buttonProps => {
+    const { shouldCloseOnButtonClick, shouldUpdateButtonValue } = this.props;
+
+    return event => {
+      if (shouldUpdateButtonValue) {
+        this.setState({ buttonValue: buttonProps.children });
+      }
+
+      if (shouldCloseOnButtonClick) {
+        this.closeDropdown();
+      }
+
+      buttonProps.handleOnClick(event, buttonProps.name);
```

Comment:
What are your thoughts about sending `buttonProps` in this callback instead of `buttonProps.name`? This way we can send all the information about the button to the parent and let them worry about how to use the information. Sending just `name` may not be flexible enough.